### PR TITLE
Add readme with implementation details

### PR DESCRIPTION
**Pull request by rubenvp8510**
Signed-off-by: Ruben Vargas <ruben.vp8510@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+This Jaeger storage plugin stores the spans using the filesystem and retrieve a completed trace using  **traceID**. 
+
+This plugin doesn't have query capabilities, the method `FindTraces(ctx context.Context, query *spanstore.TraceQueryParameters)` ignores the query parameter and return all traces stored.
+
+
+## Implementation notes
+
+###  Directory Layout
+
+Each trace has his own directory, the name of the directory is the traceID, that directory contains a list of files, each file represents and span, the name of the file is the spanID.
```
Comment (objectiser) [inline README.md]: "his" -> "its"

Comment (objectiser) [reply README.md]: "represents a span"

File: README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    ├── <traceID>
+    │   ├── <spanID>
+    │   ├── <spanID>
+    ,
+    .
+```
+
+### Read path
+
+This implementation only supports reading traces by TraceID, for each read request it list all files under `{basePath}/traces/{traceID}` and unmarshal the spans.
```
Comment (objectiser) [inline README.md]: "TraceID. For each..."   "it will list"

File: README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+In order to unmarshal the spans, it starts `N` workers and sends the span file path through channels to the workers, the workers will consume the channel, open the file and unmarshal it,  then write a result buffer. When all spans are read, the read method returns the results.
```
Comment (objectiser) [inline README.md]: should it be "through the channel to the workers"? i.e. is it actually a single channel (I haven't looked at the code yet).
If you could make that the end of the sentence and start new sentence for "The workers will consume from the channel...."
"then write" - can you remove a space preceding 'then'

File: README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+
+### Write path
+
+For write spans to the filesystem when `WriteSpan` is called, it stores the span in a buffered channel.
+
+We have `M` workers watching for the channel and writing the files to the filesystem, the span will be written to: `{basePath}/traces/{traceID}/{spanID}`.
+ 
+The span is marshaling using the proto buffer marshal method before writing to the file.
```
Comment (objectiser) [inline README.md]: "marshaling" -> "marshalled"

File: README.md (hunk 3 above)
Comment (rubenvp8510) [reply README.md]: You're right, it is only one channel.

**Review by objectiser — APPROVED** [review]

**Pull request merged**